wp/plugin/wp.ml

```diff
@@ -431,8 +431,10 @@ let callback
     })
   in
   Parameters.validate params files >>= fun params ->
-  Analysis.run params files ctxt >>= fun () ->
-  Ok ()
+  Analysis.run params files ctxt >>= function
+  | Z3.Solver.UNSATISFIABLE -> exit 0
```

Comment:
Hmm...I don't know if BAP needs to do any special cleanup after a custom command returns. If so, maybe it's not a good idea to force a premature exit here.

I wonder if there's a way to return `Error ...` with specific exit codes. Maybe a question for Ivan?

Or, maybe we could return `Ok ()` for UNSAT, `Error ...` for SAT, and then `exit 2` for UNKNOWN?

Reply:
That's a very good point. I'll take a look into it!